packages/service-library/src/servicelib/redis.py

```diff
@@ -135,7 +135,11 @@ async def _extend_lock(lock: Lock) -> None:
                 # lock is in use now
                 yield ttl_lock
         finally:
-            await ttl_lock.release()
+            # lock can expire before before being released
+            # making sure release never raises an error because the lock
+            # is no longer owned
+            with log_catch(logger, reraise=False):
```

Comment:
So you propose to suppress all exceptions raised by ``release()``.
Is there a way to verify programatically that the raised exception is because the lock is no longer owned and not any other reason?

Reply:
So what you get is the following exception: https://github.com/ITISFoundation/osparc-simcore/actions/runs/4880515069/jobs/8708953574#step:8:394 in short `LockNotOwnedError("Cannot release a lock that's no longer owned")`.
The error is raised by: `extend`, `reacquire` and `release` methods. Would you suggest to only suppress this one?

Reply:
So if I understand it well, this exception is raise because the lock was already released. Is that correct?

> Would you suggest to only suppress this one?
Yes. IMO you should suppress only what you assume is "acceptable" to ignore

Reply:
The exception is raised because the lock was not extended. The same can happen if you do not own it.
I cannot distinguish between the two above.

The reason this is happening is because it expired, since there was some delay since it was last extended and the moment it is removed. This can happen due to a lot of reasons, most notably timing.

Reply:
Will suppress exception